platform-sdk/platform-apps/tests/AddressBookTestingTool/src/main/java/com/swirlds/demo/addressbook/AddressBookTestingToolState.java

```diff
@@ -0,0 +1,253 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.demo.addressbook;
+/*
+ * This file is public domain.
+ *
+ * SWIRLDS MAKES NO REPRESENTATIONS OR WARRANTIES ABOUT THE SUITABILITY OF
+ * THE SOFTWARE, EITHER EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED
+ * TO THE IMPLIED WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE, OR NON-INFRINGEMENT. SWIRLDS SHALL NOT BE LIABLE FOR
+ * ANY DAMAGES SUFFERED AS A RESULT OF USING, MODIFYING OR
+ * DISTRIBUTING THIS SOFTWARE OR ITS DERIVATIVES.
+ */
+
+import static com.swirlds.logging.LogMarker.STARTUP;
+
+import com.swirlds.common.io.streams.SerializableDataInputStream;
+import com.swirlds.common.io.streams.SerializableDataOutputStream;
+import com.swirlds.common.merkle.MerkleLeaf;
+import com.swirlds.common.merkle.impl.PartialMerkleLeaf;
+import com.swirlds.common.system.InitTrigger;
+import com.swirlds.common.system.Platform;
+import com.swirlds.common.system.PlatformWithDeprecatedMethods;
+import com.swirlds.common.system.Round;
+import com.swirlds.common.system.SoftwareVersion;
+import com.swirlds.common.system.SwirldDualState;
+import com.swirlds.common.system.SwirldState;
+import com.swirlds.common.system.address.AddressBook;
+import com.swirlds.common.system.events.ConsensusEvent;
+import com.swirlds.common.system.transaction.ConsensusTransaction;
+import com.swirlds.common.utility.ByteUtils;
+import java.io.IOException;
+import java.time.Instant;
+import java.util.Iterator;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * State for the AddressBookTestingTool.
+ */
+public class AddressBookTestingToolState extends PartialMerkleLeaf implements SwirldState, MerkleLeaf {
+
+    private static final Logger logger = LogManager.getLogger(AddressBookTestingToolState.class);
+
+    /** modify this value to change how updateStake behaves. */
+    private int stakingProfile = 1;
+
+    private static class ClassVersion {
+        public static final int ORIGINAL = 1;
+    }
+
+    private static final long CLASS_ID = 0xf052378c7364ef47L;
+
+    private long selfId;
+
+    /**
+     * The true "state" of this app. Each transaction is just an integer that gets added to this value.
+     */
+    private long runningSum = 0;
+
+    /**
+     * The timestamp of the first event after genesis.
+     */
+    private Instant genesisTimestamp;
+
+    private boolean immutable;
```

Comment:
The abstract class `PartialMerkleLeaf` has good default implementations for immutability. Use the `setImmutable()` method when you want to make this node immutable.

Reply:
updated to setImmutable(true)